src/lib/dcob.rb

```diff
@@ -6,14 +6,33 @@
 require "pp"
 require "openssl"
 require "toml"
+require "rack"
+require "prometheus/client"
+require "prometheus/client/rack/collector"
+require "prometheus/client/rack/exporter"
 
 module Dcob
   class Server < Sinatra::Base
+    use Rack::Deflater, if: ->(env, status, headers, body) { body.any? && body[0].length > 512 }
+    use Prometheus::Client::Rack::Collector
+    use Prometheus::Client::Rack::Exporter
+
+    def initialize(app = nil)
+      super
+      prometheus = Prometheus::Client.registry
+      @no_dco_signoff = prometheus.counter(:no_dco_signoff, "The count of PRs with no DCO sign off")
+      @dco_signoff = prometheus.counter(:dco_signoff, "The count of PRs with a DCO sign off")
+      @failed_signatures ||= prometheus.counter(:failed_github_signature, "The count of failed GitHub webhook signatures")
```

Comment:
2 use `=` and 1 uses `||=`? I'd expect them to be all memoized or all direct sets, though I've never used Prometheus before, so I could be missing the nuance.